### PR TITLE
Add TranslateSpecialPath to libAddon_helper interface

### PR DESCRIPTION
**Pull request by ghost**
i addons need to store special files, return the translated value because there is currently no way for an addon to do the translation step.

### CODE REVIEW

File: xbmc/addons/binary/interfaces/api1/Addon/AddonCallbacksAddon.cpp [hunk 1]
```diff
@@ -226,15 +227,19 @@ bool CAddonCallbacksAddon::GetAddonSetting(void *addonData, const char *strSetti
 
         if (id == strSettingName && !type.empty())
         {
-          if (type == "text"     || type == "ipaddress" ||
-              type == "folder"   || type == "action"    ||
-              type == "music"    || type == "pictures"  ||
-              type == "programs" || type == "fileenum"  ||
-              type == "file"     || type == "labelenum")
+          if (type == "text"       || type == "ipaddress" ||
```
Comment (FernetMenta) [inline xbmc/addons/binary/interfaces/api1/Addon/AddonCallbacksAddon.cpp]: vertical alignment is crap. please refrain from doing so. there is always an expression that is longer than the longest one before. vertical alignment only ruins the diff.

**Comment by notspiff** [comment]
this is potentially dangerous. it can easily lead to translated paths being stored if a dev is being sloppy. i'd add the necessary helper instead.

**Comment by mapfau** [comment]
ok, thx, will change it

**Comment by mapfau** [comment]
Was changed some days ago - would be nice if it can be reviewed as inputstream.mpd currently depends on this PR. Thanx!

**Comment by notspiff** [comment]
not that my opinion is worth all that much but it is exactly what i had in
mind and code looks fine to me.

**Comment by garbear** [comment]
why not allocate on the heap like the other string functions?

**Comment by FernetMenta** [comment]
objections?

**Comment by mapfau** [comment]
> why not allocate on the heap like the other string functions?
@garbear who alloates and who disallocates? Eiher addon OR kodi should do this:
In both ways it it leads to one more call: Either Addon has to call: GetTranslatedPathLen() before (then Addon do allocatinf and deallocating, or we have to implement a FreeTransLateString() and kodi can alloc and free.
To keep it simple I used the same way as done in all addons with such strings.

**Comment by garbear** [comment]
@mapfau see how the 3 calls below it work. the add-on lib strdup's a string, and then the add-on calls FreeString(). FreeTranslateString() isn't necessary. You're correct that this requires an extra function call, but this should be taken care of in the wrapper that Alwin is writing. This way, we can have a higher-level library that returns a std::string to the add-on instead of playing buffer games.

**Comment by mapfau** [comment]
@garbear you mean there is already a freestring? wich library is it? have not seen it yet.

**Comment by garbear** [comment]
freestring() is in every file you've touched

**Comment by mapfau** [comment]
ah: DLLEXPORT void XBMC_free_string(void* hdl, void* cb, char* str)
OK, I'll change it

**Comment by mapfau** [comment]
done

**Comment by garbear** [comment]
awesome, looks good

**Comment by FernetMenta** [comment]
jenkins build this please

**Pull request merged**